crates/solidity/outputs/cargo/crate/src/generated/language.rs

```diff
@@ -7009,111 +8653,83 @@ impl Lexer for Language {
                         Some('=') => Some(TokenKind::AsteriskEqual),
                         Some(_) => {
                             input.undo();
-                            Some(TokenKind::Asterisk)
+                            None
                         }
-                        None => Some(TokenKind::Asterisk),
+                        None => None,
                     },
                     Some('+') => match input.next() {
                         Some('+') => Some(TokenKind::PlusPlus),
                         Some('=') => Some(TokenKind::PlusEqual),
                         Some(_) => {
                             input.undo();
-                            Some(TokenKind::Plus)
+                            None
                         }
-                        None => Some(TokenKind::Plus),
+                        None => None,
                     },
                     Some(',') => Some(TokenKind::Comma),
                     Some('-') => match input.next() {
                         Some('-') => Some(TokenKind::MinusMinus),
                         Some('=') => Some(TokenKind::MinusEqual),
                         Some(_) => {
                             input.undo();
-                            Some(TokenKind::Minus)
+                            None
                         }
-                        None => Some(TokenKind::Minus),
+                        None => None,
                     },
                     Some('.') => Some(TokenKind::Period),
-                    Some('/') => match input.next() {
-                        Some('=') => Some(TokenKind::SlashEqual),
-                        Some(_) => {
-                            input.undo();
-                            Some(TokenKind::Slash)
-                        }
-                        None => Some(TokenKind::Slash),
-                    },
-                    Some(':') => Some(TokenKind::Colon),
+                    Some('/') => scan_chars!(input, '=').then_some(TokenKind::SlashEqual),
```

Comment:
I see the new code is no longer producing the shorter `TokenKind::Slash`. Is this intended? How is it matched in this case?

Same thing for others like `TokenKind::Caret` and `TokenKind::GreaterThanGreaterThan`.

Reply:
The DSL v1 constructed trie from the literal scanners, however v2 used explicit `not_followed_by` rules, hence the difference. The The former relied more on the trie; the latter uses the negative lookahead and relies more on the longest match to properly tokenize.

I haven't made a decision what to use, since I needed the v1 mechanism to still work (as the migration was gradual and it supports parser codegen from v1 and v2 atm). I'd probably lean towards deriving the negative lookahead/trie from the literal definition, since explicit `not_followed_by` requires extra manual work and can lead to bugs when modifying the grammar later.

Reply:
Thanks for explaining. That makes sense. I hoped we can remove `not_followed_by` at some point all together, and automatically calculate it, but it is definetely not blocking or even critical right now.
I added a note to #638 to discuss it later.